### DISCUSSION TITLE
Copilot no longer works in PyCharm

### DISCUSSION THREAD

**Discussion by danhje** [community/community#20091] [Copilot Conversations]
It used to work great, but the last couple of weeks it hasn't given me a single suggestion.

What I've tried:
* Uninstalling and re-installing the plugin.
* Logging out and in again
* Disabling and enabling completions.
* Downgrading to 1.1.16.1219 (based on a suggestion in another thread)

I'm using:
* PyCharm 2021.3 Professional Edition, build #PY-213.5744.248, built on November 30, 2021.
* GitHub Copilot plugin 1.1.28.1744

![image](https://user-images.githubusercontent.com/8713259/177532061-58725e32-b9e9-4508-ba90-faa462aa31c7.png)
![image](https://user-images.githubusercontent.com/8713259/177531478-79342c5a-00e1-4fa8-af34-92726b287041.png)
![image](https://user-images.githubusercontent.com/8713259/177531594-7d754ae8-f976-4f42-93a1-93af32578230.png)
![image](https://user-images.githubusercontent.com/8713259/177531654-3bca241f-47c5-4db0-a3cb-748b292eeb36.png)
![7A21FF7F-7F3E-4074-A1E9-9D21B741126A](https://user-images.githubusercontent.com/8713259/177605160-5444112f-fc7b-4014-8afb-a24d23d0e6ac.jpeg)

**Comment by danhje** [comment]
Is there any way to get in contact with a support team?

**Reply by Griphcode** [reply]
Well you could look on github's support site for details but i'm not sure if it will state anything about them removing copilot from PyCharm

[Github support](https://support.github.com/)

**Comment by davecheney** [comment]
@danhje i see no requests for your user account in the last 7 days. Is it possible that your employer, internet service provider, or government, may be intercepting your SSL traffic?

**Reply by danhje** [reply]
I've been on holiday for the last 7 days. 

My employer does intercept SSL traffic, but I have no problem logging out and back in. And Copilot worked perfectly until it didn't, which was around the time the public beta ended and I signed up for the trial period.

I guess it is possible that some firewall rule was changed, and only API calls not related to authentication are effected. Any way confirm or rule that out?

**Reply by danhje** [reply]
Update: I guess you were right. Found the following in the log:

```
2022-07-31 21:48:35,966 [ 947401]   INFO -                       #copilot - [INFO] [default] [2022-07-31T19:48:35.965Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
2022-07-31 21:48:36,251 [ 947686]   WARN -                       #copilot - [ERROR] [default] [2022-07-31T19:48:36.250Z] Request Error: self signed certificate in certificate chain
2022-07-31 21:48:36,252 [ 947687]   WARN -                       #copilot - [ERROR] [ghostText] [2022-07-31T19:48:36.251Z] Error on ghost text request FetchError: self signed certificate in certificate chain
```

Not sure why this wasn't an issue previously. Anyways, I have tried to add my organizations root CA to the list of trusted certificates in Settings -> Tools - > Server Certificates, but no change. Does GitHub Copilot not respect these settings?

**Reply by danhje** [reply]
@davecheney it would be great if you could point me in the right direction. How do I tell the plug-in to trust my organization's certificate?

**Reply by davecheney** [reply]
> Does GitHub Copilot not respect these settings?

Currently it does not.

**Reply by joshystuart** [reply]
Will it? Or will I have to cancel my paid sub now?

**Reply by danhje** [reply]
I'm wondering about the same thing, @davecheney

**Reply by davecheney** [reply]
We've recently added proxy support to VSCode and Intellij. If your employer requires a proxy for general internal access, please give https://github.com/orgs/community/discussions/29127 a shot.

**Reply by danhje** [reply]
Thanks. I don't think my organization requires a proxy, so I don't think that helps me. I get access to the internet, it's just that certificates are replaced in order to be able to do deep package inspection.

**Reply by davecheney** [reply]
I know this is also something the client developers are looking into, but I don't have anything to share at the moment.

**Reply by joshystuart** [reply]
Yeah, Im in the same position as @danhje; my org deep inspects.

Copilot worked previously after adding the certificates directly to intellij/webstorm/etc, but it doesn't work now. 

Im able to run `curl --verbose https://copilot-proxy.githubusercontent.com/_ping` fine, but I can see the cert is replaced by my orgs one (which confirms the deep inspection).

My trial is up in a couple of days, so Ill have to cancel it until this is resolved Thanks.

**Reply by danhje** [reply]
Would be nice if the client devs could give an ETA for this. In the meantime, are there anyone who can help come up with a workaround? I know VS Code users have been able to get Copilot to work by setting rejectUnauthorized=false in the extension.js file. I've tried doing the same in service.js and agent.js in the folder for the IntelliJ Copilot plugin, without success.

**Comment by nlist2** [comment]
Likely because copilot is no longer free (?)

**Reply by danhje** [reply]
The public beta is over, but there's still a free trial.

**Comment by wheech-lv** [comment]
I can't use the plugin anymore either. It asks me, again and again, to authorize it on the github instead of any informative errors.

**Reply by davecheney** [reply]
@wheech-lv 
Please try visiting https://github.com/settings/copilot and completing the required setup steps.

**Comment by LanDeQuHuXi** [comment]
We desperately need a fix for PyCharm. Even all the workarounds are no longer working now, e.g. [here](https://github.com/orgs/community/discussions/8866#discussioncomment-2868122)

**Comment by LanDeQuHuXi** [comment]
Does someone know how we can reach out to the team, this is a paid product now, we need to know whether this is gonna get solved or not?

**Comment by vyshnevskyi** [comment]
Still doesn't work in pycharm due to self signed certificate. Is there any ETA on fix or at least make plugin respect IDE's Server Certificates settings?

**Comment by siddharthgoel88** [comment]
I found a solution for this which works for me. I have blogged about it at  https://sidd.io/2023/01/github-copilot-self-signed-cert-issue/ . Since Intellij and PyCharm both are by JetBrains and have same architecture so this should work for both.

Fix is as follows :

1. Export the self-signed certificate in discussion
2. Convert it into `.pem` format if not already
3. Export the path of this `.pem` cert to `NODE_EXTRA_CA_CERTS` variable
4. Restart your Intellij/PyCharm and it should work

**Reply by danhje** [reply]
Thanks for sharing your solution! I won't be able to test this on PyCharm at the moment, but if anyone else can confirm it I will mark as correct answer.
